addons/metadata.integral.artists/lib/scraper.py

```diff
@@ -0,0 +1,355 @@
+            listitem.setProperty('artist.genre', item['genre'])
+            listitem.setProperty('artist.born', item['born'])
+            listitem.setProperty('relevance', item['relevance'])
+            url = {'artist':item['artist']}
+            if 'mbid' in item:
+                url['mbid'] = item['mbid']
+            if 'dcid' in item:
+                url['dcid'] = item['dcid']
+            xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=json.dumps(url), listitem=listitem, isFolder=True)
+
+    def return_nfourl(self, item):
+        url = {'artist':item['artist'], 'mbid':item['mbartistid']}
+        listitem = xbmcgui.ListItem(offscreen=True)
+        xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=json.dumps(url), listitem=listitem, isFolder=True)
+
+    def return_resolved(self, item):
+        url = {'artist':item['artist'], 'mbid':item['mbartistid']}
+        listitem = xbmcgui.ListItem(path=json.dumps(url), offscreen=True)
+        xbmcplugin.setResolvedUrl(handle=int(sys.argv[1]), succeeded=True, listitem=listitem)
+
+    def return_details(self, item):
+        listitem = xbmcgui.ListItem(item['artist'], offscreen=True)
+        if 'mbartistid' in item:
+            listitem.setProperty('artist.musicbrainzid', item['mbartistid'])
+        if 'genre' in item:
+            listitem.setProperty('artist.genre', item['genre'])
+        if 'biography' in item:
+            listitem.setProperty('artist.biography', item['biography'])
+        if 'styles' in item:
+            listitem.setProperty('artist.styles', item['styles'])
+        if 'moods' in item:
+            listitem.setProperty('artist.moods', item['moods'])
+        if 'instruments' in item:
+            listitem.setProperty('artist.instruments', item['instruments'])
+        if 'disambiguation' in item:
+            listitem.setProperty('artist.disambiguation', item['disambiguation'])
+        if 'type' in item:
+            listitem.setProperty('artist.type', item['type'])
+        if 'sortname' in item:
+            listitem.setProperty('artist.sortname', item['sortname'])
+        if 'active' in item:
+            listitem.setProperty('artist.years_active', item['active'])
+        if 'born' in item:
+            listitem.setProperty('artist.born', item['born'])
+        if 'formed' in item:
+            listitem.setProperty('artist.formed', item['formed'])
+        if 'died' in item:
+            listitem.setProperty('artist.died', item['died'])
+        if 'disbanded' in item:
+            listitem.setProperty('artist.disbanded', item['disbanded'])
+        art = {}
+        if 'logo' in item:
+            art['logo'] = item['logo']
+        if 'banner' in item:
+            art['banner'] = item['banner']
+        if 'clearart' in item:
+            art['clearart'] = item['clearart']
+        if 'landscape' in item:
+            art['landscape'] = item['landscape']
+        listitem.setArt(art)
```

Comment:
Is the addon limiting what art types get returned from those in the results returned from the sites? Is that needed?

Reply:
please elaborate... i'm not sure i understand what you mean

Reply:
... it currently returns only 1 clearart image, instead of all available clearart images (if that's what you mean)

i fixed that a few days ago, but hadn't updated this PR with those changes.

Reply:
Yes returning only 1 image of each type was part of it, I was also unclear that it was saving all the aspects (except for fanart) in `artist.thumbs` property not just _thumb_

Also for art types other than thumb and fanart the art of that type for the artist is being set to the first fetched, overwriting any image that user may have chosen. I'm not sure that the scraper should be setting art, and if it does then only filling the gaps not overwriting chosen art.

Reply:
Checked and this is causing scraping to overwrite previously chosen art. Filling the gaps, as happens for thumbs and fanart, is done by the scraper core code. All the scraper itself has to do is fill the artist.thumbs property, listitem.setArt(art) is not wanted.

Reply:
@DaveTBlake i've removed listitem.setArt(art) from the scraper and scanned a few albums to the library.
but the result is that lots of art is now missing in the skin (clearart, banner, back, spine, ...).
these art types are also no longer listed in the choose art dialog.

Reply:
@ronie do you have the extended artwork types configured in [advancedsettings.xml](https://kodi.wiki/view/Advancedsettings.xml#Music_library_extra_artwork)? That should handle all artwork assigned.

For art types not listed in the choose art dialog, that's because there aren't any assigned in your library yet - possibly because of the above. I've got a change in the works to add all available image types to the list; I should have a PR this week.